google/cloud/storage/client.h

```diff
@@ -1646,6 +1646,52 @@ class Client {
                                std::string{}, std::forward<Options>(options)...)
         .Result();
   }
+
+  /**
+   * Restores a soft-deleted object.
+   *
+   * When a soft-deleted object is restored, a new copy of that object is
+   * created in the same bucket and inherits the same metadata as the
+   * soft-deleted object. The inherited metadata is the metadata that existed
+   * when the original object became soft deleted, with the following
+   * exceptions:
+   *
+   * 1. The createTime of the new object is set to the time at which the
+   * soft-deleted object was restored.
+   * 2. The softDeleteTime and hardDeleteTime values are cleared.
```

Comment:
nit: Should the fields `softDeleteTime` and `hardDeleteTime` be snaked_cased ?

Reply:
I don't think so. This is how the JSON API would return these fields (and how they are documented).